docs/jitsi-dev.md

```diff
@@ -0,0 +1,90 @@
+in the react-sdk), though for some widgets special integration can be done. v2 widgets
+have a `data` object which helps achieve that special integration, though v1 widgets
+are best iframed and left alone.
+
+Widgets have a `postMessage` API they can use to interact with Riot, which also allows
+Riot to interact with them. Typically this is most used by the sticker picker (an
+account-level widget), though widgets like the Jitsi widget will request permissions to
+get 'stuck' into the room list during a conference.
+
+Widgets can be added with the `/addwidget <url>` command.
+
+## Brief introduction to integration managers
+
+Integration managers (like Scalar and Dimension) are accessible via the 4 squares in
+the top right of the room and provide a simple UI over top of bridges, bots, and other
+stuff to plug into a room. They are a separate service to Riot and are thus iframed
+in a dialog as well. They also have a `postMessage` API they can use to interact with
+the client to create things like widgets, give permissions to bridges, and generally
+set everything up for the integration the user is working with.
+
+Integration managers do not currently have a spec associated with them, though efforts
+are underway in [MSC1286](https://github.com/matrix-org/matrix-doc/issues/1286).
+
+## Widgets configured by integration managers
+
+Integration managers will often "wrap" a widget by using a widget `url` which points
+to the integration manager instead of to where the user requested the widget be. For
+example, a custom widget added in an integration manager for https://matrix.org will
+end up creating a widget with a URL like `https://integrations.example.org?widgetUrl=https%3A%2F%2Fmatrix.org`.
+
+The integration manager's wrapper will typically have another iframe to isolate the
+widget from the client by yet another layer. The wrapper often provides other functionality
+which might not be available on the embedded site, such as a fullscreen button or the
+communication layer with the client (all widgets *should* be talking to the client
+over `postMessage`, even if they aren't going to be using the widget APIs).
+
+Widgets added with the `/addwidget` command will *not* be wrapped as they are not going
+through an integration manager.
+
+## Jitsi widgets from integration managers
+
+Integration managers will create an entire widget event and send it over `postMessage`
+for the client to add to the room. This means that the integration manager gets to 
+decide the conference domain, conference name, and other aspects of the widget. As
+a result, users can end up with a Jitsi widget that does not use the same conference
+server they specified in their config.json - this is expected.
+
+Some integration managers allow the user to change the conference name while others
+will generate one for the user. 
+
+## Jitsi widgets generated by Riot itself
+
+When the user clicks on the call buttons by the composer, the integration manager is
```

Comment:
Perhaps it's useful to clarify that the call buttons only trigger a widget when the room has more than 2 members. For just 2 members, a direct WebRTC call is made.